R/utilities-simulation-results.R

```diff
@@ -18,14 +18,18 @@
 #' @param population population used to calculate the simulationResults (optional). This is used only to add the population covariates to the resulting data table.
 #'
 #' @param stopIfNotFound Boolean. If TRUE and no result exist for the given path, an error is thrown. Default is \code{TRUE}
-#' @param stopIfNotFound If \code{TRUE} (default) an error is thrown if no results exist for any `path`
-#' If \code{FALSE}, a list of \code{NA} values is returned for the repsecitve path.
+#' @param stopIfNotFound If \code{TRUE} (default) an error is thrown if no results exist for any `path`. If \code{FALSE}, a list of \code{NA} values is returned for the respective path.
+#' @param withMetaData If \code{TRUE} (default), the output is a list two sublists `data`and
+#' `metaData`, with latter storing information about units and dimensions of the outputs. If \code{FALSE}, \code{metaData} is \code{NULL}. Setting this option to \code{FALSE} might improve
+#' the performance of the function.
+#'
 #' @export
 getOutputValues <- function(simulationResults,
                             quantitiesOrPaths = NULL,
                             population = NULL,
                             individualIds = NULL,
-                            stopIfNotFound = TRUE) {
+                            stopIfNotFound = TRUE,
+                            withMetaData = TRUE) {

```

Comment:
I really don't like this. Return value structure driven by input parameters. It feels wrong to me.
Why do you need this? I thought the changes brought down the time to almost nothing.

Reply:
For the tested use-case, not returning the meta data (resp. returning NULL for meta data) reduces the time from 1 sec to 500 msec.

I am not sure, but I would say for >90% of the use cases this meta data is not used, and in performance-critical scenarios (e.g. PI) this argument would allow to almost completely avoid communication with .NET. With the default `TRUE`, the code that is already present would behave exactly as it did.

I must admit, I never understood why this meta-data is required in the first place. If it is for TLF, why cannot TLF use ospsuite-r for this?

Reply:
THe idea is that you can plot your outputs in a one liner in TLF if you know what entities you are dealing with (dimension and unit for instance). I can leave with this micro optimization but in general, we need to be mindful that those input driven return values are VERY hard to comprehend. This is typically a big no no and I am not sure if this is something specific to R or not.

Reply:
I agree that in general this is not a good concept. Just in this case it is a feature that costs performance in a potentially performance-sensitive scenario, while it is not used in >90% of cases. I cannot imagine why someone would plot simulation results without loading ospsuite-r.

For TLF, a method that adds simulation results to a DataMapping can be implemented that queries the units paths, like

`DataMapping$addSimulationResults(simulationResults, paths)`